### PR TITLE
Issue 6266: Fixing the bookkeeper image spec in the system test framework

### LINKED ISSUE #6266: System Test Framework does not use the right bookkeeper image
**Describe the bug**
System Test Framework does not use the right image repository and pull policy for the bookkeeper image, and therefore ends up using the default values for those fields.

**To Reproduce**
Trigger a system test run on Jenkins with the latest bookkeeper operator image and a bookkeeper image in a repository other than `pravega/bookkeeper`

**Screenshots**
N.A.

**Additional information**
The issue is observed while using the latest bookkeeper operator (above version 0.1.2) and a bookkeeper image which is not in the default repository (i.e. `pravega/bookkeeper`)

### PR DESCRIPTION
**Pull request by SrishT**
Signed-off-by: SrishT <Srishti.Thakkar@dell.com>

**Change log description**  
This PR modifies the image specification within the bookkeeper cluster spec, so that it reflects the right directory structure.

**Purpose of the change**  
Fixes #6266 

**What the code does**  
Correctly populates the bookkeeper image spec.

**How to verify it**  
Triggering a system test run on Jenkins with the latest bookkeeper operator image and a bookkeeper image in a repository other than the default repository (i.e. `pravega/bookkeeper`) should be able to point to the desired bookkeeper image.

### CODE REVIEW

File: test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java [hunk 1]
```diff
@@ -251,6 +251,13 @@ public String getID() {
                 .build();
     }
 
+    protected Map<String, Object> getBookkeeperImageSpec(String imageName) {
```
Comment (anishakj) [inline test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java:254]: Cant we make this method private?

**Review by anishakj — CHANGES_REQUESTED** [review]

File: test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java [hunk 2]
```diff
@@ -334,7 +341,7 @@ private V1ConfigMap getBookkeeperOperatorConfigMap() {
     private Map<String, Object> getBookkeeperDeployment(String zkLocation, int bookieCount, ImmutableMap<String, String> props) {
         // generate BookkeeperSpec.
         final Map<String, Object> bkPersistentVolumeSpec = getPersistentVolumeClaimSpec("10Gi", "standard");
-        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder().put("image", getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + BOOKKEEPER_IMAGE_NAME, BOOKKEEPER_VERSION))
+        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder().put("image", getBookkeeperImageSpec(DOCKER_REGISTRY + PREFIX + "/" + BOOKKEEPER_IMAGE_NAME))
```
Comment (RaulGracia) [inline test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java:344]: Is this change compatible with existing versions of Bookkeeper Operator?

Comment (SrishT) [reply test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java:344]: No Raul, these changes are only compatible with bookkeeper operator versions above 0.1.3.
But this change would help us migrate to the latest bookkeeper operator versions.

File: test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java (hunk 1 above)
Comment (SrishT) [reply test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java]: done

**Review by anishakj — APPROVED** [review]
LGTM

**Review by RaulGracia — APPROVED** [review]
Do we need to describe somewhere that now system tests only work with Bookkeeper Operator 1.3+?

**Review by shrids — APPROVED** [review]
+1

**Pull request merged**